File: Tests/VB/TestAllDatabases/TestAllDatabases/TestFixtures/NamingFixture.vb
'======================================
' Copyright EntitySpaces, LLC 2005 - 2006 
'======================================

Imports System.Data

Imports NUnit.Framework
'using Adapdev.UnitTest;
Imports BusinessObjects

Namespace Tests.Base
	<TestFixture> _
	Public Class NamingFixture
		Private namingTestColl As New NamingTestCollection()
		Private namingTest As New NamingTest()

		<TestFixtureSetUp> _
		Public Sub Init()
		End Sub

		<SetUp> _
		Public Sub Init2()
			namingTestColl = New NamingTestCollection()
			namingTest = New NamingTest()
			' The table Naming.Test has a 'dot' in it,
			' so that is being tested as well.
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTestColl.LoadAll()
					namingTestColl.MarkAllAsDeleted()
					namingTestColl.Save()
					Exit Select
				Case Else

					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithSpaceAndAlias()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTest = namingTestColl.AddNew()
					namingTest.str.TestAliasSpace = "AliasSpace"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithDot()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTest = namingTestColl.AddNew()
					namingTest.str.TestFieldDot = "FieldDot"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithDotAndSetGuids()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					Dim keyGuid As Guid = Guid.NewGuid()
					namingTest = namingTestColl.AddNew()
					namingTest.GuidKeyAlias = keyGuid
					namingTest.str.TestFieldDot = "FieldDot"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Assert.AreEqual(namingTestColl(0).GuidKeyAlias.Value, keyGuid)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithUnderScoreAndAlias()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTest = namingTestColl.AddNew()
					namingTest.str.Test_Alias_Underscore = "AliasUnder"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithSpaceWithoutAlias()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTest = namingTestColl.AddNew()
					namingTest.str.TestFieldSpace = "FieldSpace"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub ColumnWithUnderScoreWithoutAlias()
			Select Case namingTestColl.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					namingTest = namingTestColl.AddNew()
					namingTest.str.TestFieldUnderscore = "FieldUnder"
					namingTestColl.Save()
					Assert.IsTrue(namingTestColl.LoadAll())
					Assert.AreEqual(1, namingTestColl.Count)
					Exit Select
				Case Else

					Assert.Ignore("Database not set up, yet.")
					Exit Select
			End Select
		End Sub
	End Class
End Namespace

File: Tests/CSharp/TestAllDatabases/TestAllDatabases/TestFixtures/NamingFixture.cs
//======================================
// Copyright EntitySpaces, LLC 2005 - 2006 
//======================================

using System;
using System.Data;

using NUnit.Framework;
//using Adapdev.UnitTest;
using BusinessObjects;

namespace Tests.Base
{
	[TestFixture]
	public class NamingFixture
	{
		NamingTestCollection namingTestColl = new NamingTestCollection();
        NamingTest namingTest = new NamingTest();
		
		[TestFixtureSetUp]
		public void Init()
		{
        }

		[SetUp]
		public void Init2()
		{
			namingTestColl = new NamingTestCollection();
            namingTest = new NamingTest();
            // The table Naming.Test has a 'dot' in it,
            // so that is being tested as well.
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTestColl.LoadAll();
                    namingTestColl.MarkAllAsDeleted();
                    namingTestColl.Save();
                    break;

                default:
                    break;
            }
        }

		[Test]
		public void ColumnWithSpaceAndAlias()
		{
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTest = namingTestColl.AddNew();
					namingTest.str().TestAliasSpace = "AliasSpace";
					namingTestColl.Save();
					Assert.IsTrue(namingTestColl.LoadAll());
					Assert.AreEqual(1, namingTestColl.Count);
					break;

				default:
					Assert.Ignore("Database not set up, yet.");
					break;
			}
		}

        [Test]
        public void ColumnWithDot()
        {
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTest = namingTestColl.AddNew();
                    namingTest.str().TestFieldDot = "FieldDot";
                    namingTestColl.Save();
                    Assert.IsTrue(namingTestColl.LoadAll());
                    Assert.AreEqual(1, namingTestColl.Count);
                    break;

                default:
                    Assert.Ignore("Database not set up, yet.");
                    break;
            }
        }

        [Test]
        public void ColumnWithDotAndSetGuids()
        {
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    Guid keyGuid = Guid.NewGuid();
                    namingTest = namingTestColl.AddNew();
                    namingTest.GuidKeyAlias = keyGuid;
                    namingTest.str().TestFieldDot = "FieldDot";
                    namingTestColl.Save();
                    Assert.IsTrue(namingTestColl.LoadAll());
                    Assert.AreEqual(1, namingTestColl.Count);
                    Assert.AreEqual(namingTestColl[0].GuidKeyAlias.Value, keyGuid);
                    break;

                default:
                    Assert.Ignore("Database not set up, yet.");
                    break;
            }
        }

        [Test]
		public void ColumnWithUnderScoreAndAlias()
		{
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTest = namingTestColl.AddNew();
                    namingTest.str().Test_Alias_Underscore = "AliasUnder";
                    namingTestColl.Save();
                    Assert.IsTrue(namingTestColl.LoadAll());
                    Assert.AreEqual(1, namingTestColl.Count);
                    break;

                default:
                    Assert.Ignore("Database not set up, yet.");
                    break;
            }
		}

		[Test]
		public void ColumnWithSpaceWithoutAlias()
		{
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTest = namingTestColl.AddNew();
					namingTest.str().TestFieldSpace = "FieldSpace";
					namingTestColl.Save();
					Assert.IsTrue(namingTestColl.LoadAll());
					Assert.AreEqual(1, namingTestColl.Count);
					break;

				default:
					Assert.Ignore("Database not set up, yet.");
					break;
			}
		}

		[Test]
		public void ColumnWithUnderScoreWithoutAlias()
		{
            switch (namingTestColl.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
                    namingTest = namingTestColl.AddNew();
					namingTest.str().TestFieldUnderscore = "FieldUnder";
					namingTestColl.Save();
					Assert.IsTrue(namingTestColl.LoadAll());
					Assert.AreEqual(1, namingTestColl.Count);
					break;

				default:
					Assert.Ignore("Database not set up, yet.");
					break;
			}
		}
    }
}
